runtime/rust/python-wheel/tests/soak.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import random
+import string
+
+import uvloop
+from triton_distributed_rs import DistributedRuntime, triton_worker
+
+# Soak Test
+#
+# This was a failure case for the distributed runtime. If the Rust Tokio
+# runtime is started with a small number of threads, it will starve the
+# the GIL + asyncio event loop can starve timeout the ingress handler.
+#
+# There may still be some blocking operations in the ingress handler that
+# could still eventually be a problem.
+
+
+@triton_worker()
+async def worker(runtime: DistributedRuntime):
+    ns = random_string()
+    task = asyncio.create_task(server_init(runtime, ns))
+    await client_init(runtime, ns)
+    runtime.shutdown()
+    await task
+
+
+async def client_init(runtime: DistributedRuntime, ns: str):
+    """
+    Instantiate a `backend` client and call the `generate` endpoint
+    """
+    # get endpoint
+    endpoint = runtime.namespace(ns).component("backend").endpoint("generate")
+
+    # create client
+    client = await endpoint.client()
+
+    # wait for an endpoint to be ready
+    await client.wait_for_endpoints()
+
+    # issue 1000 concurrent requests
+    # the task should issue the request and process the response
+    tasks = []
+    for i in range(5000):
```

Comment:
Swapping 1000 for "many" so it doesn't have to be in sync with the code below (5000)
```suggestion
    # Issue many concurrent requests to load the server.
    # The task should issue the request and process the response
    tasks = []
    for i in range(5000):
```